stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java

```diff
@@ -24,4 +31,56 @@ public void consume(final StockItem stockItem) {
 
         stockItemRepository.addOrReplace(stockItem);
     }
+
+    @KafkaListener(topics = {OrdersTopics.ORDER_CHECKED_OUT})
+    public void consumeOrderCheckedOut(final Order order)
+            throws ResourceNotFoundException, InsufficientStockException, InvalidStockChangeException {
+        logger.info(String.format("#### -> Consumed message -> %s", order));
+
+        // TODO: lock the stock to prevent multiple checks on same stock
+        // check availability of all StockItems
+        Iterator<UUID> stockItemIdIterator = order.getItemIds().iterator();
+
+        while (stockItemIdIterator.hasNext()) {
+            StockItem stockItem;
+            try {
+                stockItem = stockItemRepository.find(stockItemIdIterator.next());
+            } catch (ResourceNotFoundException e) {
+                producer.emitStockItemsSubtractForOrderFailed(order);
+                return;
+            }
+
+            if (stockItem.getStock() < 1) { // amount per order item is 1
+                producer.emitStockItemsSubtractForOrderFailed(order);
+                return;
+            }
+        }
+
+        // all stocks available, proceed and emit a successful event
+        int totalPrice = 0;
+        stockItemIdIterator = order.getItemIds().iterator();
+
+        while(stockItemIdIterator.hasNext()) {
+            StockItem stockItem;
+            try {
+                stockItem = stockItemRepository.find(stockItemIdIterator.next());
+            } catch (ResourceNotFoundException e) {
+                // we assume this should not cause any exception
+                throw e;
+            }
+
+            try {
+                stockItem.subtractStock(1);
+            } catch (InsufficientStockException | InvalidStockChangeException e) {
+                // we assume the subtraction should not cause any exception
+                throw e;
+            }
+
+            totalPrice += stockItem.getPrice() * 1;
```

Comment:
Why is there a ```* 1``` here?

Reply:
The amount of the stock item

Reply:
But thats always 1, why should there be a ```* 1``` then?

Reply:
Done :)